### ISSUE TITLE
Deferred Components Google play saying One or more of the auto-generated multi-APKs exceeds the maximum allowed size of 150 MB.

### ISSUE THREAD

**Issue by copypasteearth** [flutter/flutter#100855]
## Steps to Reproduce

I have 300MB bundle and I made a Deferred Component to hold 150MB worth of assets mp3 files.  I uploaded it to google play and it keeps telling me One or more of the auto-generated multi-APKs exceeds the maximum allowed size of 150 MB.

in my main.dart I have a FutureBuilder that shows a loading indicator while i call the future          [DeferredComponent.installDeferredComponent(componentName: "meditations")]

I tried to follow the steps exactly and I just cannot get it to work.  I will provide whatever details you need, please this is imperative, I have to have this release by April 1st.

Thank you for your time.

Am I going about this wrong, any help to get this done would be great

**Comment by copypasteearth** [comment]
@maheshmnj so is there a way to use deferred components as an asset pack?

**Comment by maheshmnj** [comment]
@copypasteearth Unfortunately flutter currently doesn't support [Play Asset Delivery](https://developer.android.com/guide/playcore/asset-delivery). Deferred components is based on [Play feature delivery](https://developer.android.com/guide/playcore/feature-delivery). Labeling this issue as a proposal to Add support for Play Asset Delivery to by pass the size restriction limit when deploying to playstore.

**Comment by maheshmnj** [comment]
However, you might want to checkout third party plugins from pub  https://pub.dev/packages/flutter_play_asset to unblock yourself.

**Comment by maheshmnj** [comment]
cc: @GaryQian

**Comment by GaryQian** [comment]
Since we don't control the android/play store side limitations, have you considered breaking up the assets into more components. If you break up into 3 deferred components, then each is ~100mb well under the limit. Since you are so close to the limit, different ways of counting mb (eg, 1000 vs 1024) may be putting you over somewhere

**Comment by GaryQian** [comment]
Could your issue be related to https://github.com/flutter/flutter/issues/96046 ?

If so, you can try using an older version of Flutter as this is a more recent regression.

**Comment by copypasteearth** [comment]
@GaryQian I tried flutter 2.2.3 and 2.2.5 I think, and the latest, I wound up using the play_asset_delivery package on pub.dev I noticed a difference was that I put "install-time" when using it in the build.gradle of the asset pack would that have made it work with deferred components?

**Comment by GaryQian** [comment]
Using install-time would make the assets all download at initial install, so you would loose the benefit of dynamically downloading as needed, but it would indeed work in getting around the size limit.

**Comment by shapljts** [comment]
Any update on this? This is pretty critical, especially for offline-first and other data-intensive apps. It would be ideal to have Play Asset Delivery support in the platform.

**Comment by pyrossh** [comment]
Seems like there is this library now which helps in asset delivery. 
https://pub.dev/packages/flutter_play_asset_delivery/example

But it would be good if we get official support as all other platforms allow upto 2GB for app sizes. But doesn't seem like there is a straight forward way.

**Comment by vandir** [comment]
Any update on this?

**Comment by denghejun** [comment]
any update? need a solution

**Comment by untillnesss** [comment]
any update? need a solution

**Comment by Soap-141** [comment]
Would love this feature

**Comment by Gilianp** [comment]
Any updates?

**Comment by yamashita-room-335** [comment]
Using the Flutter and [pigeon](https://pub.dev/packages/pigeon) library, I have created a sample app that can use the Play Asset Delivery (Android) and On-Demand Resources (iOS) features.

https://github.com/yamashita-room-335/online_assets

I have included a fair amount of comments as readme, so you may be able to implement these features in your own apps by copying any implementation you like!

**Comment by simeonApproppo** [comment]
I just found a easy way, how to integrate `install-time` asset packs with Play Asset Delivery.
Just add a Asset-Pack Folder in your android-Directory according https://developer.android.com/guide/playcore/asset-delivery/integrate-native (build.gradle for your module, add assetPacks on your app/build.gradle and add the include in the settings.gradle)
**Then ensure that you put your assets under `android/<myAssetPack>/src/main/assets/flutter_assets/assets/<your-assets>`. Then you can access your assets in a normal way like: `Image.asset('assets/<your-assets>')`**

This only works with `install-time` asset packs and when you deliver your AAB over Google Play. For local-testing and iOS just add your assets directly in the default-asset path and uncomment the assets in the `pubspec.yaml` when you make your Play-build.

With this you can deliver 4GB app bundles to Google Play (like you already can with IPAs on Apple App Store).
(Each asset pack should not exceed 1.5GB.)
